Jenkinsfile

```
@@ -55,35 +55,53 @@ node('Slave') {
-                reportDir: 'reports/coverage/ruby',
-                reportFiles: 'index.html',
-                reportName: 'Ruby Code Coverage'
-            ])
+                publishHTML (target: [
+                    allowMissing: false,
+                    alwaysLinkToLastBuild: false,
+                    keepAll: true,
+                    reportDir: 'reports/coverage/ruby',
+                    reportFiles: 'index.html',
+                    reportName: 'Ruby Code Coverage'
+                ])
+            }
+        } 
+        catch(e) {
+            pipelineStatus = 'FAILED'
+            currentBuild.result = 'FAILURE'
         }
 
-        emailext (
-            to: emailList,
-            subject: "${pipelineStatus}: Job '${env.JOB_NAME} [${env.BUILD_NUMBER}]' in stage ${curStage}",
-            body: """<p>${pipelineStatus}: Job '${env.JOB_NAME} [${env.BUILD_NUMBER}]' in stage '${curStage}' for branch '${branch}':</p>
-            <p>Check console output at &QUOT;<a href='${env.BUILD_URL}'>${env.JOB_NAME} [${env.BUILD_NUMBER}]</a>&QUOT;</p>"""
-        )
-        slackSend (color: '#FF0000', message: "${pipelineStatus}: Job '${env.JOB_NAME} [${env.BUILD_NUMBER}]' in stage '${curStage}' for branch '${branch}' (${env.BUILD_URL})")
+        try {
+            emailext (
+                to: emailList,
+                subject: "${pipelineStatus}: Job '${env.JOB_NAME} [${env.BUILD_NUMBER}]' in stage ${curStage}",
+                body: """<p>${pipelineStatus}: Job '${env.JOB_NAME} [${env.BUILD_NUMBER}]' in stage '${curStage}' for branch '${branch}':</p>
+                <p>Check console output at &QUOT;<a href='${env.BUILD_URL}'>${env.JOB_NAME} [${env.BUILD_NUMBER}]</a>&QUOT;</p>"""
+            )
+
+            slackAlertColor = '#11AB1B'
+            slackMessage = "${pipelineStatus}: Job '${env.JOB_NAME} [${env.BUILD_NUMBER}]' completed for branch '${branch}' (${env.BUILD_URL})"
+
+            if(pipelineStatus == 'FAILED') {
```

Instead of SUCCESS as default and then if FAILURE, could be non-success as default and then if SUCCESS, this would cover null, false, 127, etc if something went extra wrong.